src/Xamarin.Android.Build.Tasks/Tasks/ClassParse.cs

```diff
@@ -26,43 +20,33 @@ public class ClassParse : AndroidTask
 
 		public ITaskItem [] DocumentationPaths { get; set; }
 
-		public override bool RunTask ()
+		protected override string GenerateCommandLineCommands ()
 		{
-			using (var output = new StreamWriter (OutputFile, append: false, 
-						encoding: Files.UTF8withoutBOM)) {
-				Bytecode.Log.OnLog = LogEventHandler;
-				var classPath = new Bytecode.ClassPath () {
-					ApiSource = "class-parse",
-					DocumentationPaths = (DocumentationPaths ?? Enumerable.Empty<ITaskItem> ()).Select(x => x.ItemSpec)
-				};
-				foreach (var jar in SourceJars) {
-					if (Bytecode.ClassPath.IsJarFile (jar.ItemSpec)) {
-						classPath.Load (jar.ItemSpec);
-					}
-				}
-				classPath.SaveXmlDescription (output);
+			var cmd = GetCommandLineBuilder ();
+
+			var responseFile = Path.Combine (Path.GetDirectoryName (OutputFile), "class-parse.rsp");
```

Comment:
If `class-parse.exe` succeeds, should we delete this file immediately? Just leave it on disk if there is a non-zero exit code?

Otherwise, should something delete it when you run the `Clean` target?

Reply:
Good suggestions, add performance data and `<FileWrites />` for response file.